src/workerd/api/node/util.c++

```diff
@@ -240,4 +241,39 @@ jsg::JsValue UtilModule::getBuiltinModule(jsg::Lock& js, kj::String specifier) {
   return js.undefined();
 }
 
+void UtilModule::processExitImpl(jsg::Lock& js, int code) {
+  if (IoContext::hasCurrent()) {
+    // There are a few things happening here. First, we abort the current IoContext
+    // in order to shut down this specific request....
+    auto message =
+        kj::str("The Node.js process.exit(", code, ") API was called. Canceling the request.");
+    auto& ioContext = IoContext::current();
+    // If we have a tail worker, let's report the error.
+    KJ_IF_SOME(tracer, ioContext.getWorkerTracer()) {
+      auto ex = KJ_ASSERT_NONNULL(js.error(message).tryCast<jsg::JsObject>());
+      tracer.addException(ioContext.now(), ex.get(js, "name"_kj).toString(js),
+          ex.get(js, "message"_kj).toString(js), ex.get(js, "stack"_kj).toString(js));
+      ioContext.abort(js.exceptionToKj(ex));
+    } else {
+      ioContext.abort(JSG_KJ_EXCEPTION(FAILED, Error, kj::mv(message)));
+    }
+    // ...then we tell the isolate to terminate the current JavaScript execution.
+    // Oddly however, this does not appear to *actually* terminate the thread of
+    // execution unless we trigger the Isolate to handle the intercepts, which
+    // calling v8::JSON::Stringify does. Weird... but ok? As long as it works
+    // TODO(soon): Investigate if there is a better approach to triggering the
+    // interrupt handling.
+    js.v8Isolate->TerminateExecution();
```

Comment:
Does `IoContext::runImpl()` do the right thing after this? It looks like it'll throw a KJ exception `script terminated for unknown reasons` which would end up in our Sentry.

Reply:
What do you suggest as an alternative?

Reply:
The one idea I've come up with is checking immediately before the `script terminated for unknown reasons` assert to see if the `abortFulfiller` is still waiting. If it is not, then we assume that the script terminated because the IoContext was aborted and we emit a `NOSENTRY` error instead.

Reply:
Actually, would it actually? The `process.exit(...)` impl does not actually throw a `jsg::JsExceptionThrown` in this case that I can see. I'm not getting that log in my testing on this.

Reply:
In any case, let's go with this and if it ends up being a problem in sentry we can revisit. Use of `process.exit(...)` *should* be limited and I kind of do want to know if folks are using it.

Reply:
> The one idea I've come up with is checking immediately before the script terminated for unknown reasons assert to see if the abortFulfiller is still waiting. If it is not, then we assume that the script terminated because the IoContext was aborted and we emit a NOSENTRY error instead.

That sounds fine. We can also rethrow the abort exception from runImpl() in this case.

Don't log NOSETRY, though. Just don't log anything, because there was no error. (If you separately want to log NOSENTRY to find out if people are using process.exit(), then do that in process.exit()'s own implementation... but I don't like that kind of logging, I worry it'll never get cleaned up and will just waste log space.)

> The process.exit(...) impl does not actually throw a jsg::JsExceptionThrown in this case that I can see. I'm not getting that log in my testing on this.

Err, is it actually terminating execution or not? Hard to see how it wouldn't end up throwing if it's correctly terminating execution...

Reply:
Added the jsg::check and logic to verify that that execution is terminating. Still not actually seeing the "script terminated for unknown reasons" log in the tests.

Reply:
What if you KJ_DBG() next to the line where that message is thrown?

I am guessing that the error is being thrown but the abort error is winning the race out of WorkerEntrypoint so the "script terminated for unknown reasons" error is discarded. But that might not be reliably true...

Reply:
Added the additional check here. Haven't been able to trigger this assert, however, so unsure how necessary it is.